src/matrix/net/HomeServerRequest.ts

```diff
@@ -30,13 +31,18 @@ export class HomeServerRequest implements IHomeServerRequest {
     // as we add types for expected responses from hs, this could be a generic class instead
     private readonly _promise: Promise<any>;
 
-    constructor(method: string, url: string, sourceRequest: RequestResult, log?: ILogItem) {
+    constructor(method: string, url: string, sourceRequest: RequestResult, options?: IRequestOptions) {
```

Comment:
`IRequestOptions` are the options for the `HomeServerAPI._requestFn`, should we be using it here?

Reply:
No, the options passed to the different methods on HomeServerApi is actually a different object, the typing seems wrong. Well spotted. This object has log, timeout, uploadProgress and prefix (which we should not put on there really, but should just be an argument of _get, _post, ...). At the same time, `IRequestOptions` does not have log or prefix. Can you add a type and fix it? :pray: